src/Command/Util/StyleCiPatchCommand.php

```diff
@@ -0,0 +1,110 @@
+use Gush\Helper\DownloadHelper;
+use Gush\Helper\GitHelper;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class StyleCiPatchCommand extends BaseCommand implements GitRepoFeature, GitFolderFeature
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('pull-request:styleci')
+            ->setDescription('Run StyleCI patches on given PR')
+            ->addArgument('pr_number', InputArgument::REQUIRED, 'PR number')
+            ->setHelp(
+                <<<EOF
+The <info>%command.name%</info> command applies StyleCI patches on given PR:
+
+    <info>$ gush %command.name% 12</info>
+
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $adapter = $this->getAdapter();
+
+        if (!$adapter instanceof GitHubAdapter) {
+            throw new UserException('Usage of StyleCI is currently limited to the GitHub adapter.');
+        }
+
+        $org = $input->getOption('org');
+        $repo = $input->getOption('repo');
+
+        $prNumber = $input->getArgument('pr_number');
+        $pr = $adapter->getPullRequest($prNumber);
+
+        /** @var GitHelper $gitHelper */
+        $gitHelper = $this->getHelper('git');
+
+        if (!$gitHelper->isWorkingTreeReady()) {
+            throw new WorkingTreeIsNotReady();
+        }
+
+        $status = $this->getStatus($adapter->getCommitStatuses($org, $repo, $pr['head']['ref']));
+
+        if ('success' === $status['status']) {
+            $this->getHelper('gush_style')->error('Nothing to update.');
+
+            return self::COMMAND_SUCCESS;
+        }
+
+        /** @var DownloadHelper $downloadHelper */
+        $downloadHelper = $this->getHelper('download');
+        $patchFile = $downloadHelper->downloadFile($status['patch_url']);
+
+        $patchOperation = $gitHelper->createRemotePatchOperation();
+        $patchOperation->setRemote($pr['head']['user'], $pr['head']['ref']);
+        $patchOperation->applyPatch($patchFile, 'correct CS', 'p1');
+        $patchOperation->pushToRemote();
+
+        $this->getHelper('gush_style')->success('StyleCI patch was applied and pushed.');
+
+        return self::COMMAND_SUCCESS;
+    }
+
+    private function getStatus(array $statuses)
+    {
+        foreach ($statuses as $status) {
+            if (false !== stripos($status['context'], 'StyleCI')) {
+                return [
+                    'status' => $status['state'],
+                    'patch_url' => $status['target_url'].'/diff'
+                ];
+            }
+        }
+
+        throw new UserException(
+            'No StyleCI status found in commit, make sure StyleCI is enabled for the repository.'."\n".
+            'And that analyzes exist for the commit/pull-request.'
```

Comment:
Shouldn't this be `analyses` instead of `analyzes `?
Also there should be a comma on the previous line to use `and` (and without uppercase `A`)